crates/starknet-devnet/tests/test_blocks_on_demand.rs

```diff
@@ -273,12 +302,31 @@ mod blocks_on_demand_tests {
             tx_hashes.push(invoke_result.transaction_hash);
         }
 
+        assert_eq!(
+            get_contract_balance_by_block_id(
+                &devnet,
+                contract_address,
+                BlockId::Tag(BlockTag::Pending)
```

Comment:
it's handled by `blocks_on_demand_states_and_blocks` but it's done by mint there and here we use invoke so I decided to add it anyways

Reply:
oh sorry, now I remember that it can't be asserted from the latest block in this case :D

since the new block is not created contract only exists in the pending block/state so I'm getting Call failed: ContractNotFound

Reply:
no problem, add an assertion that the contract is not found

Reply:
added in d691b08f0ba4c9b99b67f88dba1e561039387fba